File: Bookshop/Bookshop/Models/Review.swift
//
//  SimpleOrder.swift
//  BookShop
//
//  Created by dennis savchenko on 09/06/2025.
//

import Foundation
import SwiftUI

/// Represents a customer review for an item.
struct Review: Identifiable, Decodable {
    let id: Int
    let text: String
    let rating: Int
    let timeStamp: Date
    let username: String
    let itemId: Int
    let customerId: Int
}

File: BookshopServer/Entities/Review.cs
namespace BookShopServer.Entities;

/// <summary>
/// Represents a customer review of an item in the bookshop system,
/// including rating, text content, timestamp, and references to the customer and item.
/// </summary>
public class Review
{
    public int Id { get; set; }
    public int CustomerId { get; set; }
    public int ItemId { get; set; }
    public int Rating { get; set; }
    public string Text { get; set; }
    public DateTime TimeStamp { get; set; }
    public virtual Customer Customer { get; set; }
    public virtual Item Item { get; set; }
}
